tensorboard/backend/event_processing/event_accumulator_test.py

```diff
@@ -29,895 +29,915 @@
+        self.assertEqual(acc.FirstEventTimestamp(), 10)
+
+    def testReloadPopulatesFirstEventTimestamp(self):
+        """Test that Reload() means FirstEventTimestamp() won't load events."""
+        gen = _EventGenerator(self)
+        acc = ea.EventAccumulator(gen)
+        gen.AddEvent(
+            tf.Event(wall_time=1, step=2, file_version='brain.Event:2'))
+
+        acc.Reload()
+
+        def _Die(*args, **kwargs):  # pylint: disable=unused-argument
+            raise RuntimeError('Load() should not be called')
+
+        self.stubs.Set(gen, 'Load', _Die)
+        self.assertEqual(acc.FirstEventTimestamp(), 1)
+
+    def testFirstEventTimestampLoadsEvent(self):
+        """Test that FirstEventTimestamp() doesn't discard the loaded event."""
+        gen = _EventGenerator(self)
+        acc = ea.EventAccumulator(gen)
+        gen.AddEvent(
+            tf.Event(wall_time=1, step=2, file_version='brain.Event:2'))
+
+        self.assertEqual(acc.FirstEventTimestamp(), 1)
+        acc.Reload()
+        self.assertEqual(acc.file_version, 2.0)
+
+    def testTFSummaryScalar(self):
+        """Verify processing of tf.summary.scalar."""
+        event_sink = _EventGenerator(self, zero_out_timestamps=True)
+        writer = tf.summary.FileWriter(self.get_temp_dir())
+        writer.event_writer = event_sink
+        with self.test_session() as sess:
+            ipt = tf.placeholder(tf.float32)
+            tf.summary.scalar('scalar1', ipt)
+            tf.summary.scalar('scalar2', ipt * ipt)
+            merged = tf.summary.merge_all()
+            writer.add_graph(sess.graph)
+            for i in xrange(10):
+                summ = sess.run(merged, feed_dict={ipt: i})
+                writer.add_summary(summ, global_step=i)
+
+        accumulator = ea.EventAccumulator(event_sink)
+        accumulator.Reload()
+
+        seq1 = [ea.ScalarEvent(wall_time=0, step=i, value=i) for i in
+                xrange(10)]
+        seq2 = [
+            ea.ScalarEvent(
+                wall_time=0, step=i, value=i * i) for i in xrange(10)
+        ]
+
+        self.assertTagsEqual(accumulator.Tags(), {
+            ea.SCALARS: ['scalar1', 'scalar2'],
+            ea.GRAPH: True,
+            ea.META_GRAPH: False,
+        })
+
+        self.assertEqual(accumulator.Scalars('scalar1'), seq1)
+        self.assertEqual(accumulator.Scalars('scalar2'), seq2)
+        first_value = accumulator.Scalars('scalar1')[0].value
+        self.assertTrue(isinstance(first_value, float))
+
+    def testTFSummaryImage(self):
+        """Verify processing of tf.summary.image."""
+        event_sink = _EventGenerator(self, zero_out_timestamps=True)
+        writer = tf.summary.FileWriter(self.get_temp_dir())
+        writer.event_writer = event_sink
+        with self.test_session() as sess:
+            ipt = tf.ones([10, 4, 4, 3], tf.uint8)
+            # This is an interesting example, because the old
+            # tf.image_summary op
```

Comment:
Another example where comments get hard-wrapped individually instead of as blocks.